src/main/java/org/jabref/logic/citationstyle/CitationStyleCache.java

```diff
@@ -1,58 +1,61 @@
 package org.jabref.logic.citationstyle;
 
-import java.util.HashMap;
-import java.util.Map;
 import java.util.Objects;
 
 import org.jabref.model.database.BibDatabaseContext;
 import org.jabref.model.database.event.EntryRemovedEvent;
 import org.jabref.model.entry.BibEntry;
 import org.jabref.model.entry.event.EntryChangedEvent;
 
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
 import com.google.common.eventbus.Subscribe;
 
 
 /**
  * Caches the generated Citations for quicker access
- * {@link CitationStyleGenerator} generates the citaiton with JavaScript which may take some time
+ * {@link CitationStyleGenerator} generates the citation with JavaScript which may take some time
  */
 public class CitationStyleCache {
 
-    private CitationStyle citationStyle = CitationStyle.getDefault();
-    private Map<BibEntry, String> citationStylesCache = new HashMap<>();
+    private static final int CACHE_SIZE = 1024;
+    private CitationStyle citationStyle;
+    private LoadingCache<BibEntry, String> citationStyleCache;
 
 
     public CitationStyleCache(BibDatabaseContext bibDatabaseContext) {
         this(bibDatabaseContext, CitationStyle.getDefault());
     }
 
     public CitationStyleCache(BibDatabaseContext bibDatabaseContext, CitationStyle citationStyle) {
-        this.setCitationStyle(citationStyle);
+        this.citationStyle = citationStyle;
+        citationStyleCache = CacheBuilder.newBuilder().maximumSize(CACHE_SIZE).build(new CacheLoader<BibEntry, String>() {
+            @Override
+            public String load(BibEntry entry) throws Exception {
```

Comment:
According to https://stackoverflow.com/a/34424577/873282, this is not possible. The basic class has more than one abstract method (see https://github.com/google/guava/blob/master/guava/src/com/google/common/cache/CacheLoader.java#L127).